### PR TITLE
ported to mingw (mxe/gcc10)

### PR DESCRIPTION
**Pull request by luntik2012**
network and types ifdef-s added for `__linux__` and `_WIN32` (no `OSX`), linked with `ws2_32`  for `WIN32`.

not tested in real applications.

didn't try to build on windows/mingw (i'll do it soon)

### CODE REVIEW

**Comment by jadamroth** [comment]
Thanks for submitting. 

I'll test sometime this week on a real end device to make sure it still compiles and works properly on linux, mac, and windows

**Comment by jadamroth** [comment]
I noticed this fork a while ago where someone also implemented windows support, but they never submitted a pull request, and I never reviewed the code - https://github.com/shrddr/EIPScanner

I don't know if you referenced it or not, but it might be helpful.

**Comment by jadamroth** [comment]
Just got around to testing this.

I can confirm it works on linux (ubuntu 18.04), still need to test on windows. There's a lot of compile and library issues on OSX that need to be resolved before I can merge this pull request.

- [X] Linux
- [ ] Windows
- [ ] OSX

**Comment by jadamroth** [comment]
If you give me write permissions to your forked repo, I can pushed my fixed OSX changes where I can confirm it works on linux and osx. 

Then if my changes still work for you on windows, I'll merge the pull request

**Comment by jadamroth** [comment]
Added commits for linux and macos. Please test windows

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -55,6 +55,32 @@ set(SOURCE_FILES
 add_library(EIPScanner SHARED ${SOURCE_FILES})
 add_library(EIPScannerS STATIC ${SOURCE_FILES})
 
+message(${PROJECT_SOURCE_DIR}/src)
```
Comment (jadamroth) [inline src/CMakeLists.txt:58]: This should not be included in the pull request. This looks like debug logging

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -55,6 +55,32 @@ set(SOURCE_FILES
...
+
+target_include_directories(EIPScanner PUBLIC
```
Comment (jadamroth) [inline src/CMakeLists.txt:60]: We already use include_directories(${CMAKE_SOURCE_DIR}/src) in our root CMakeLists.txt. All of the include paths of the project should be relative

**Review by jadamroth — CHANGES_REQUESTED** [review]
Are you linking this library to another project or are you using it standalone as in the examples? 

I would assume that you are linking it. If so, you should move your include files to the necessary directories and link from your main project rather than adding these commits to the repo. However, the best method would be to use cmake's ExternalProject.

Unless for some additional reason I don't understand, I cannot accept these recent commits.

File: src/CMakeLists.txt (hunk 1 above)
Comment (luntik2012) [reply src/CMakeLists.txt:58]: I link it to my project and I had some troubles with includes. How should I try to modify CMakeLists.txt?

File: src/CMakeLists.txt (hunk 2 above)
Comment (luntik2012) [reply src/CMakeLists.txt:60]: > I noticed this fork a while ago where someone also implemented windows support, but they never submitted a pull request, and I never reviewed the code - https://github.com/shrddr/EIPScanner
> 
> I don't know if you referenced it or not, but it might be helpful.

I've looked at this. Changes are almost the same, except I didn't touch whitespace, didn't change c++ version and didn't implemented/changed any new logic. Also, it's important to fix SO_RCVTIMEO usage for windows. Today I've tested it on windows and it's stable.

**Comment by luntik2012** [comment]
> I noticed this fork a while ago where someone also implemented windows support, but they never submitted a pull request, and I never reviewed the code - https://github.com/shrddr/EIPScanner
> 
> I don't know if you referenced it or not, but it might be helpful.

I've looked at this. Changes are almost the same, except I didn't touch whitespace, didn't change c++ version and didn't implemented/changed any new logic. Also, it's important to fix SO_RCVTIMEO usage for windows. Today I've tested it on windows and it's stable.

**Comment by jadamroth** [comment]
Thanks, I've reviewed changes. 

Give me a few days, and I'll test with an end device

**Comment by jadamroth** [comment]
From my initial tests, everything appears to be working on both linux and mac. I'm unable to test with windows + mingw, so I'll assume from your replies that everything's working as it's supposed to be.

I'm going to merge this pull request into master and make a minor release (version 1.2.0). Any additional necessary changes will be in patch releases from new issues and pull requests.

Let me know if you run in to any other issues. I'll continue testing with macos and linux

Thanks for your contribution and using the library.
